sim/dex-data.ts

```diff
@@ -102,14 +114,11 @@ export class BasicEffect implements EffectData {
 	sourceEffect: string;
 
 	constructor(data: AnyObject) {
-		this.exists = true;
-		Object.assign(this, data);
-
 		this.name = Utils.getString(data.name).trim();
 		this.id = data.realMove ? toID(data.realMove) : toID(this.name); // Hidden Power hack
 		this.fullname = Utils.getString(data.fullname) || this.name;
 		this.effectType = Utils.getString(data.effectType) as EffectType || 'Condition';
-		this.exists = !!(this.exists && this.id);
+		this.exists = data.exists ?? !!this.id;
 		this.num = data.num || 0;
 		this.gen = data.gen || 0;
```

Comment:
It occurs to me that if you want the common prefix as long as possible, `fullname` and `num` should be below `gen`, to match `TypeInfo`.